x-pack/plugins/uptime/public/lib/helper/component_test_helpers.tsx

```diff
@@ -0,0 +1,187 @@
+import { coreMock } from 'src/core/public/mocks';
+import {
+  KibanaContextProvider,
+  KibanaServices,
+} from '../../../../../../src/plugins/kibana_react/public';
+import { MountWithReduxProvider } from './helper_with_redux';
+import { AppState } from '../../state';
+
+interface KibanaProps {
+  services?: KibanaServices;
+}
+
+interface KibanaProviderOptions {
+  coreOptions?: any;
+  kibanaProps?: KibanaProps;
+}
+
+interface MockKibanaProviderProps extends KibanaProviderOptions {
+  children: ReactElement;
+}
+
+interface MockRouterProps extends MockKibanaProviderProps {
+  history?: History;
+}
+
+interface RenderRouterOptions extends KibanaProviderOptions {
+  history?: History;
+  renderOptions?: any;
+  state?: any;
+}
+
+/* default mock core */
+const defaultCore = coreMock.createStart();
+const mockCore: () => any = () => {
+  const core = {
+    ...defaultCore,
+    application: {
+      getUrlForApp: () => '/app/uptime',
+      navigateToUrl: jest.fn(),
+    },
+  };
+
+  return core;
+};
+
+/* Higher Order Components */
+export function withKibanaContext<T>(
+  WrappedComponent: React.ComponentType<T>,
+  { kibanaProps, coreOptions }: KibanaProviderOptions
+) {
+  const core = {
+    ...mockCore(),
+    coreOptions,
+  };
+  return (props: any) => (
+    <KibanaContextProvider services={{ ...core }} {...kibanaProps}>
+      <I18nProvider>
+        <WrappedComponent {...(props as T)} />
+      </I18nProvider>
+    </KibanaContextProvider>
+  );
+}
+
+export function withRouter<T>(WrappedComponent: React.ComponentType<T>, customHistory: History) {
+  const history = customHistory || createMemoryHistory();
+  return (props: T) => (
+    <Router history={history}>
+      <WrappedComponent {...(props as T)} />
+    </Router>
+  );
+}
+
+/* Mock Provider Components */
+export function MockKibanaProvider({
+  children,
+  coreOptions,
+  kibanaProps,
+}: MockKibanaProviderProps) {
+  const core = {
+    ...mockCore(),
+    coreOptions,
+  };
+  return (
+    <KibanaContextProvider services={{ ...core }} {...kibanaProps}>
+      <I18nProvider>{children}</I18nProvider>
+    </KibanaContextProvider>
+  );
+}
+
+export function MockRouter({
+  children,
+  coreOptions,
+  history: customHistory,
+  kibanaProps,
+}: MockRouterProps) {
+  const history = customHistory || createMemoryHistory();
+  return (
+    <Router history={history}>
+      <MockKibanaProvider coreOptions={coreOptions} kibanaProps={kibanaProps}>
+        {children}
+      </MockKibanaProvider>
+    </Router>
+  );
+}
+
+export const render = (
+  ui: ReactElement,
+  { history, coreOptions, kibanaProps, renderOptions, state }: RenderRouterOptions = {}
+) => {
+  return reactRender(
+    <MountWithReduxProvider store={state}>
+      <MockRouter history={history} kibanaProps={kibanaProps} coreOptions={coreOptions}>
+        {ui}
+      </MockRouter>
+    </MountWithReduxProvider>,
+    renderOptions
+  );
+};
+
+/* Enzyme helpers */
```

Comment:
These old helpers look like a direct copy/paste from the previous file, am I right? If modified I'll look more closely.

Reply:
They are copied.